### PR TITLE
introduce data workspace extension

### PR DESCRIPTION
**Pull request by alanrenmsft**
I'd like to merge this into main so that we can work in parallel.

1. extension folder structure
1. projects view container and projects view
1. add project button
1. for now the extension will not ship with ADS
image

@SakshiS-harma , you can implement the projects.addProject command in main.ts to open the dialog that you are about to work on.

### CODE REVIEW

File: extensions/data-workspace/.vscodeignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+coverageConfig.json
+coverage
+out
+src
```
Comment (aaomidi) [inline extensions/data-workspace/.vscodeignore:4]: Why??

File: extensions/data-workspace/.vscodeignore [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+tsconfig.json
```
Comment (aaomidi) [inline extensions/data-workspace/.vscodeignore:5]: Why??

Comment (aaomidi) [reply extensions/data-workspace/.vscodeignore:5]: Actually most of this file, why?

File: extensions/data-workspace/package.json [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  "dependencies": {
+    "vscode-nls": "^4.0.0"
+  },
+  "devDependencies": {
+    "@types/mocha": "^5.2.5",
+    "mocha": "^5.2.0",
+    "mocha-junit-reporter": "^1.17.0",
+    "mocha-multi-reporters": "^1.1.7",
+    "typemoq": "^2.1.0",
+    "vscodetestcover": "^1.0.9",
```
Comment (aaomidi) [inline extensions/data-workspace/package.json:82]: this has updated to 1.1.0 now

File: extensions/data-workspace/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (aaomidi) [inline extensions/data-workspace/tsconfig.json:1]: Turn on strict compiling for new projects

File: extensions/data-workspace/.vscodeignore (hunk 2 above)
Comment (alanrenmsft) [reply extensions/data-workspace/.vscodeignore:5]: the vscodeignore file is used for vsce package command, decides which files will be included in the vsix file. I've updated it and removed out from this file.

File: extensions/data-workspace/tsconfig.json (hunk 4 above)
Comment (alanrenmsft) [reply extensions/data-workspace/tsconfig.json:1]: it has been specified in the shared.tsconfig.json which is being referenced here

File: extensions/data-workspace/.vscodeignore (hunk 2 above)
Comment (aaomidi) [reply extensions/data-workspace/.vscodeignore:4]: Hmm don’t those files also impact the search or am I mistaken?

Comment (Charles-Gagnon) [reply extensions/data-workspace/.vscodeignore:4]: @aaomidi No - that's a separate VS Code setting https://github.com/microsoft/azuredatastudio/blob/main/.vscode/settings.json#L16

**Comment by udeeshagautam** [comment]
@alanrenmsft just to confirm this is our final icon - yeah?

**Comment by udeeshagautam** [comment]
also looks like one of the test failed for one PR build...

**Comment by alanrenmsft** [comment]
> also looks like one of the test failed for one PR build...

@udeeshagautam it doesn't seem to be related to my changes, I have restarted the failed job.

**Comment by alanrenmsft** [comment]
> @alanrenmsft just to confirm this is our final icon - yeah?

@udeeshagautam , no, this is the bdc icon used for placeholder, do we have the actual icon now?

File: extensions/data-workspace/package.json [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    "microsoft.mssql"
+  ],
+  "contributes": {
+    "commands": [
+      {
+        "command": "projects.addProject",
+        "title": "%add-project-command%",
+        "category": "",
+        "icon": {
+          "light": "images/new.svg",
```
[the remark this answers is not in the document]
Comment (alanrenmsft) [reply extensions/data-workspace/package.json:33]: I took a look at the documentation for command contribution point, it doesn't seem to support specifying icon in this way.

File: extensions/data-workspace/src/main.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
```
Comment (alanrenmsft) [inline extensions/data-workspace/src/main.ts:1]: oops, i didn't mean to commit these changes, i 'll revert my last commit

File: extensions/data-workspace/package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (Charles-Gagnon) [reply extensions/data-workspace/package.json:33]: Yeah it does 

https://github.com/microsoft/azuredatastudio/blob/main/extensions/git/package.json#L44
https://github.com/microsoft/azuredatastudio/blob/main/extensions/arc/package.json#L46

[the remark this answers is not in the document]
Comment (alanrenmsft) [reply extensions/data-workspace/package.json:33]: wow, great to know, i am more than happy to get rid of the icon files. LOL

**Pull request merged**